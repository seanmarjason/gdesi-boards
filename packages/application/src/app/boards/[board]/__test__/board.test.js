import { render, screen } from '@testing-library/react'
import Board from '../board';

const data = [
    {id: 1, name: 'To Do', cards: []},
    {id: 2, name: 'Doing', cards: []},
    {id: 3, name: 'Done', cards: []},
]

global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(data),
    }),
  );


 
test('Board renders correctly', async () => {
  const { container } = render(<Board />)
  await screen.findByText("Your Team's Work")

  expect(container).toMatchSnapshot()
})
