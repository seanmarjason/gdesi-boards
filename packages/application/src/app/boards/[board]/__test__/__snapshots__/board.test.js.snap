// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div>
  <main
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiStack-root css-vsjeqg-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1sd6chr-MuiStack-root"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1d2i4tu-MuiTypography-root"
              >
                Boards
              </p>
            </li>
          </ol>
        </nav>
        <div
          class="MuiStack-root css-1oxswqj-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            id=":r0:"
            tabindex="0"
            type="button"
          >
            Sign out
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 css-1hd3wng-MuiTypography-root"
        >
          Your Team's Work
        </h2>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mns5ui-MuiButtonBase-root-MuiButton-root"
          href="/boards/undefined/tasks/new"
          id=":r1:"
          tabindex="0"
        >
          Create New Task
        </a>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1a6tk71-MuiContainer-root"
        >
          <div />
        </div>
      </div>
    </div>
  </main>
</div>
`;
